### PR TITLE
Upgraded VSCode Extension API + Typescript + Fixed CI Build

### PR DESCRIPTION
**Pull request by alankyshum**
Will wait for build -> See if the tests are fixed

### CODE REVIEW

**Comment by alankyshum** [comment]
Issue: 
Able to run using launch.json, but in CI, it raises an error

```
TypeError: Cannot read property 'extensionPath' of undefined
```

**Comment by ziyasal** [comment]
Hi @alankyshum, is this ready to merge?

**Comment by alankyshum** [comment]
Sorry about the delay, I was typing to fix the tests before merging in this PR, but haven't got time recently. Will followup ASAP if there's no other concerns.

And thanks for checking!

**Comment by alankyshum** [comment]
Great! At least the CI could pick up the extension! Now I'm fixing the test

**Comment by alankyshum** [comment]
Build OS X Issue fixed, now working on Linux build

**Comment by alankyshum** [comment]
TLDR; 
```
clearClipboard(); // otherwise, clipboard has nothing, and causes target string to be undefined
originalClipboardContent = paste();
```
